vpc/service/assign_addresses_v3.go

```diff
@@ -1061,40 +1060,24 @@ func assignArbitraryIPv6AddressV3(ctx context.Context, tx *sql.Tx, branchENI *ec
 	logger.G(ctx).WithField("usedIPAddresses", usedIPAddresses.List()).WithField("allInterfaceIPAddresses", allInterfaceIPAddresses.List()).Debug("Trying to assign IPv6 Address")
 	unusedIPAddresses := allInterfaceIPAddresses.Difference(usedIPAddresses)
 
-	if unusedIPAddresses.Len() > 0 {
+	if l := unusedIPAddresses.Len(); l > 0 {
 		unusedIPv6AddressesList := unusedIPAddresses.List()
 
-		rows, err := tx.QueryContext(ctx, "SELECT ip_address, last_seen FROM ip_last_used_v3 WHERE host(ip_address) = any($1) AND vpc_id = $2", pq.Array(unusedIPv6AddressesList), aws.StringValue(branchENI.VpcId))
-		if err != nil {
-			err = status.Error(codes.Unknown, errors.Wrap(err, "Could not fetch utilized IPv4 addresses from the database").Error())
-			span.SetStatus(traceStatusFromError(err))
+		row := tx.QueryRowContext(ctx, "SELECT ip_address FROM ip_last_used_v3 WHERE ip_address = any($1::inet[]) AND vpc_id = $2 ORDER BY last_seen ASC LIMIT 1", pq.Array(unusedIPv6AddressesList), aws.StringValue(branchENI.VpcId))
```

Comment:
Just to confirm, this query picks the oldest ipv6 out of the database that has been last_seen, that happens to be already assigned to this agent, just not used by any task?

Reply:
Correct.